### PR TITLE
[WIP] Travis to check deno2

### PR DESCRIPTION
**Pull request by ry**
Plan:
* Travis to only check deno2
* Once that's working remove original deno and move deno2 to root
* Leave original deno in a branch and link to it in a readme.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,36 +1,38 @@
...
+ - export PATH=$PATH:$DEPOT_TOOLS_PATH
+ - cd deno2
+ - gclient sync --no-history
+ # ccache needs this
+ - export CCACHE_CPP2=yes
+ - export CCACHE_SLOPPINESS=time_macros
+ - export PATH=`pwd`/third_party/llvm-build/Release+Asserts/bin:$PATH
+ - (cd js; yarn)
+ - rm -f $BUILD_PATH/args.gn # Don't use cached values.
+ - gn gen $BUILD_PATH --args='is_debug=false cc_wrapper="ccache" use_custom_libcxx = false use_sysroot = false'
```
Comment (piscisaureus) [inline .travis.yml]: spaces around the equals sign (`use_sysroot = false`). That works?

File: deno2/.gn [hunk 2]
```diff
@@ -8,7 +8,7 @@ buildconfig = "//build/config/BUILDCONFIG.gn"
 # The secondary source root is a parallel directory tree where
 # GN build files are placed when they can not be placed directly
 # in the source tree, e.g. for third party source trees.
-secondary_source = "//v8/"
+#secondary_source = "//v8/"
```
Comment (piscisaureus) [inline deno2/.gn]: Why is this commented out?

File: deno2/js/package.json [hunk 3]
```diff
@@ -4,7 +4,13 @@
...
     "parcel-bundler": "^1.8.1",
     "protobufjs": "^6.8.6",
-    "typescript": "^2.9.1"
+    "typescript": "^2.9.1",
+
+    "jsdoc": "^3.5.5",
+    "tmp": "0.0.33",
+    "uglify-js": "^2.8.29",
+    "espree": "^3.5.3",
+    "minimist": "^1.2.0"
```
Comment (piscisaureus) [inline deno2/js/package.json]: Why are all these extra packages needed?

File: deno2/.gn [hunk 4]
```diff
@@ -17,19 +17,41 @@ secondary_source = "//v8/"
...
   v8_deprecation_warnings = false
   v8_embedder_string = "-deno"
   v8_enable_gdbjit = false
   v8_enable_i18n_support = false
+  v8_experimental_extra_library_files = []
+  v8_extra_library_files = []
   v8_imminent_deprecation_warnings = false
-  v8_monolithic = false
+  v8_monolithic = true
+  v8_static_library=false
```
Comment (piscisaureus) [inline deno2/.gn]: Does this mean we're building a shared library? Or does v8_monolithic turn it into yet something else?

Comment (ry) [reply deno2/.gn]: Removed. Everything is statically linked in the end - this is inconsequential to our task

File: .travis.yml (hunk 1 above)
Comment (ry) [reply .travis.yml]: Fixed.

File: deno2/.gn (hunk 2 above)
Comment (ry) [reply deno2/.gn]: Removed completely. Symlinks are introduced here that circumvent the need for this.

**Comment by ry** [comment]
Supports deno_rs now and added comments.
PTAL @piscisaureus

**Pull request merged**